doc/modules/ROOT/pages/parsing.adoc

```diff
@@ -0,0 +1,37 @@
+:doctype: book
```

Comment:
@radcortez IMHO we need to drop this page, it provides some internal details which we should be free to change.
Perhaps a short intro to `JsonWebToken` with the class showing it injected can be added instead (example, simple service which returns a principal name by calling `jwt.getName()`)

Reply:
I copied this from the existent documentation:
https://github.com/smallrye/smallrye-jwt/blob/6390becaa77ece913a365fcaf05c8d469caf0dfd/docs/src/main/asciidoc/index.adoc

Since this is specific SmallRye documentation, I see no harm in providing documentation on the internal, even if they change later (as long as we update the documentation :)).

Reply:
@radcortez I did not see it, it is an outdated doc, https://github.com/smallrye/smallrye-jwt lists those properties, and `requireiss` was left out intentionally. That parsing info is irrelevant to the users who work with JsonWebToken soi needs to be dropped IMHO

Reply:
Very well, I'll remove it.

Reply:
@radcortez since then a parser interface has also been introduced, so at the moment it is a bit of mute point as to how the implementations integrate it. My understanding that none of the integrators are  dealing directly with it like this, Thorntail has a custom extension but as far as I recall it just depends on the producer to have `JsonWebToken` injected. Perhaps later on we can come up with the integration advice section and list few ideas. Lets review it later, perhaps some of this parse text can return, please open an issue so that we can discuss the ideas there. Thanks